frontend/src/product/Products.component.tsx

```diff
@@ -1,70 +1,115 @@
-import { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import { useMutation } from '@tanstack/react-query';
 import axios from 'axios';
 
-import { apiEndpoints } from '../constants/constants';
 import { useTokenContext } from '../context/TokenContext';
-import type { Element } from '../interfaces/Element.component';
+import { Element } from '../interfaces/Element.component';
 import { toastError, toastSuccess } from '../notification/ToastNotification.component';
+import { PaginationComponent } from '../pagination/Pagination.component';
+import { SearchBar } from '../searchBar/searchBar.component';
 import { ProductCard } from './ProductCard.component';
 import styles from './product-list.module.css';
 
+const PRODUCTS_PER_PAGE = 16;
+
 export const ProductsComponent: React.FC = () => {
   const [elements, setElements] = useState<Element[]>([]);
   const [loading, setLoading] = useState<boolean>(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [hasNextPage, setHasNextPage] = useState<boolean>(false);
+  const [isOpen, setIsOpen] = useState(false);
 
   const { accessToken } = useTokenContext();
+  const navigate = useNavigate();
 
   const getProducts = useMutation({
-    mutationFn: async () => {
-      const response = await axios.get(apiEndpoints.products.products);
-
+    mutationFn: async ({ page, searchTerm }: { page: number; searchTerm: string }) => {
+      const response = await axios.get('http://localhost:5000/product', {
```

Comment:
Get api url from constants always. If we change the backend port for any reason, you'd need to change it everywhere in the codebase